controllers/openstackipset_controller.go

```diff
@@ -246,6 +245,199 @@ func (r *OpenStackIPSetReconciler) Reconcile(ctx context.Context, req ctrl.Reque
+
+			// Do we already have a reservation for this hostname on the network?
+			for _, reservation := range network.Status.RoleReservations[instance.Spec.RoleName].Reservations {
+				if reservation.Hostname == hostname {
+					// We also need the netmask (which is not stored in the OpenStackNet status),
+					// so we acquire it from the OpenStackIPSet spec
+					cidrPieces := strings.Split(network.Spec.Cidr, "/")
+					reservationIP = fmt.Sprintf("%s/%s", reservation.IP, cidrPieces[len(cidrPieces)-1])
+					break
+				}
+			}
+
+			// get list of all reservations
+			reservationList := []ospdirectorv1beta1.IPReservation{}
+			for _, roleReservations := range network.Status.RoleReservations {
+				reservationList = append(reservationList, roleReservations.Reservations...)
+			}
+
+			if reservationIP == "" {
+				r.Log.Info(fmt.Sprintf("Create new reservation for host %s in network %s", hostname, netName))
+
+				// No reservation found, so create a new one
+				ip, reservation, err := common.AssignIP(common.AssignIPDetails{
+					IPnet:           *cidr,
+					RangeStart:      start,
+					RangeEnd:        end,
+					RoleReservelist: network.Status.RoleReservations[instance.Spec.RoleName].Reservations,
+					Reservelist:     reservationList,
+					ExcludeRanges:   []string{},
+					Hostname:        hostname,
+					VIP:             instance.Spec.VIP,
+					Deleted:         false,
+				})
+				if err != nil {
+					r.Log.Error(err, "Failed to do ip reservation %v")
+					return err
+				}
+
+				reservationIP = ip.String()
+
+				// record the reservation on the OpenStackNet
+				// If RoleReservations status map is nil, create it
+				if network.Status.RoleReservations == nil {
+					network.Status.RoleReservations = map[string]ospdirectorv1beta1.OpenStackNetRoleStatus{}
+				}
+				network.Status.RoleReservations[instance.Spec.RoleName] = ospdirectorv1beta1.OpenStackNetRoleStatus{
+					AddToPredictableIPs: instance.Spec.AddToPredictableIPs,
+					Reservations:        reservation,
+				}
+
+			}
+
+			// add netName -> reservationIP to hostIPs map
+			hostNetworkIPs[netName] = reservationIP
+
+			err = r.Client.Status().Update(context.TODO(), network)
+			if err != nil {
+				r.Log.Error(err, "Failed to update OpenStackNet status %v")
+				return err
+			}
+
+		}
+
+		// set/update ipset host status
+		instance.Status.HostIPs[hostname] = ospdirectorv1beta1.OpenStackIPHostsStatus{
+			IPAddresses: hostNetworkIPs,
+		}
+	}
+
+	return nil
+}
+
+func (r *OpenStackIPSetReconciler) cleanupOSNetStatus(instance *ospdirectorv1beta1.OpenStackIPSet) error {
+
+	// mark entry in OSNet status as deleted
+	for _, netName := range instance.Spec.Networks {
+		network := &ospdirectorv1beta1.OpenStackNet{}
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Name: netName, Namespace: instance.Namespace}, network)
+		if err != nil {
+			if k8s_errors.IsNotFound(err) {
+				r.Log.Info(fmt.Sprintf("OpenStackNet named %s not found!", netName))
```

Comment:
Same question here regarding the `continue` consideration from above

Reply:
yes, agreed